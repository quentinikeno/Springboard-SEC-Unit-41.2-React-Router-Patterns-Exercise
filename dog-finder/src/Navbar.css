.Navbar {
	display: flex;
	justify-content: space-around;
	margin: 2rem 10rem;
	padding: 1rem;
}

.Navbar a {
	color: rgba(0, 0, 0, 0.5);
	text-decoration: none;
	transition: all ease 1s;
	font-size: 1.25rem;
}

.Navbar :is(a.active, a:hover) {
	color: black;
}
