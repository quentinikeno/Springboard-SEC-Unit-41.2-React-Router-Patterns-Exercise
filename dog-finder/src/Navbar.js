import { NavLink } from "react-router-dom";
import "./Navbar.css";

const Navbar = ({ names }) => {
	return (
		<nav className="Navbar">
			<NavLink to="/dogs">Home</NavLink>
			{names.map((name) => (
				<NavLink to={`/dogs/${name.toLowerCase()}`} key={name}>
					{name}
				</NavLink>
			))}
		</nav>
	);
};

export default Navbar;
