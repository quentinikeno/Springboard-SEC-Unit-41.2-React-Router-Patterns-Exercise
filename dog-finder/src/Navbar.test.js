import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "./Navbar";
import dogs from "./dogs";

const dogNames = dogs.map((dog) => dog.name);

it("should render without crashing", () => {
	render(
		<MemoryRouter>
			<Navbar names={dogNames} />
		</MemoryRouter>
	);
});

it("should matchSnapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<Navbar names={dogNames} />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("should have all the dogs in the document", () => {
	const { getByText } = render(
		<MemoryRouter>
			<Navbar names={dogNames} />
		</MemoryRouter>
	);
	expect(getByText("Home")).toBeInTheDocument();
	expect(getByText("Whiskey")).toBeInTheDocument();
	expect(getByText("Duke")).toBeInTheDocument();
	expect(getByText("Perry")).toBeInTheDocument();
	expect(getByText("Tubby")).toBeInTheDocument();
});
