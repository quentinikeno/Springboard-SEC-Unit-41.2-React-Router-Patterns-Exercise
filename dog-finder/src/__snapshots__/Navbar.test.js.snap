// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matchSnapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navbar"
  >
    <a
      href="/dogs"
    >
      Home
    </a>
    <a
      href="/dogs/whiskey"
    >
      Whiskey
    </a>
    <a
      href="/dogs/duke"
    >
      Duke
    </a>
    <a
      href="/dogs/perry"
    >
      Perry
    </a>
    <a
      href="/dogs/tubby"
    >
      Tubby
    </a>
  </nav>
</DocumentFragment>
`;
